lasagne/layers/noise.py

```diff
@@ -211,3 +219,76 @@ def get_output_for(self, input, deterministic=False, **kwargs):
             return input + self._srng.normal(input.shape,
                                              avg=0.0,
                                              std=self.sigma)
+
+
+class AlphaDropoutLayer(DropoutLayer):
+    """Dropout layer.
+
+    Sets values to alpha. Will also converge the remaining
+    neurons to mean 0 and a variance of 1 unit. See notes for disabling dropout
+    during testing.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        the layer feeding into this layer, or the expected input shape
+    p : float or scalar tensor
+        The probability of setting a value to zero
+    rescale : bool
+        If ``True`` (the default), scale the input by ``1 / (1 - p)`` when
+        dropout is enabled, to keep the expected output mean the same.
+    shared_axes : tuple of int
+        Axes to share the dropout mask over. By default, each value can be
+        dropped individually. ``shared_axes=(0,)`` uses the same mask across
+        the batch. ``shared_axes=(2, 3)`` uses the same mask across the
+        spatial dimensions of 2D feature maps.
+    alpha : float or SELU instance
+        The default values are fixed point solutions to equations (4) and (5)
+        in [1] for zero mean and unit variance input. The analytic expressions
+        for them are given in equation (14) also in [1].
```

Comment:
Should probably first state what role `alpha` plays for the operation of the layer. Also the reference must be given as `[1]_`, not `[1]`.